pyfr/solvers/baseadvec/kernels/transform.mako

```diff
@@ -0,0 +1,83 @@
+<%namespace module='pyfr.backends.base.makoutil' name='pyfr'/>
+
+<% t_tol = 0.99 %>
+
+// Transforms to m = [1, 0, 0]^T
+// See Moler and Hughes 1999
+<%pyfr:macro name='transform_to' params='n, u, t, off'>
+% if ndims == 2:
+    t[off + 0] =  n[0]*u[off + 0] + n[1]*u[off + 1];
+    t[off + 1] = -n[1]*u[off + 0] + n[0]*u[off + 1];
+% elif ndims == 3:
+    if (fabs(n[0]) < ${t_tol})
+    {
+        fpdtype_t h = 1/(1 + n[0]);
+
+        t[off + 0] = n[0]*u[off + 0] + n[1]*u[off + 1] + n[2]*u[off + 2];
+        t[off + 1] = - n[1]*u[off + 0] + (n[0] + h*n[2]*n[2])*u[off + 1]
```

Comment:
Avoid having negatives out front (so `-a + b`  => `b - a`). Saves an instruction or two for compilers which do not rearrange math expressions.